gempy/core/tensor/interpolator_tf.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+import tensorflow as tf
+
+
+class Interpolator(object):
+    """Class that act as:
+    copy from theano Interpolator by Miguel
+     1) linker between the data objects and the theano graph
+     2) container of theano graphs + shared variables
+     3) container of theano function
+
+    Args:
+        surface_points (SurfacePoints): [s0]
+        orientations (Orientations): [s1]
+        grid (Grid): [s2]
+        surfaces (Surfaces): [s3]
+        series (Series): [s4]
+        faults (Faults): [s5]
+        additional_data (AdditionalData): [s6]
+        kwargs:
+            - compile_theano: if true, the function is compile at the creation of the class
+
+    Attributes:
+        surface_points (SurfacePoints)
+        orientations (Orientations)
+        grid (Grid)
+        surfaces (Surfaces)
+        faults (Faults)
+        additional_data (AdditionalData)
+        dtype (['float32', 'float64']): float precision
+        theano_graph: theano graph object with the properties from AdditionalData -> Options
+        theano function: python function to call the theano code
+
+    """
+    # TODO assert passed data is rescaled
+
+    def __init__(self, surface_points: "SurfacePoints", orientations: "Orientations", grid: "Grid",
+                 surfaces: "Surfaces", series: "Series", faults: "Faults", additional_data: "AdditionalData", **kwargs):
+        # Test
+        self.surface_points = surface_points
+        self.orientations = orientations
+        self.grid = grid
+        self.additional_data = additional_data
+        self.surfaces = surfaces
+        self.series = series
+        self.faults = faults
+
+        self.dtype = additional_data.options.df.loc['values', 'dtype']
+
+        self._compute_len_series()
+
+    def _compute_len_series(self):
```

Comment:
All these properties can be separated. I will try to design something that makes sense

Reply:
Here I just copied your code, will do that after graph.py, just to make a draft to keep notes

Reply:
yes, but organizing the python code to control the different interpolators should be my task. Just copy as much as you need for the development of the graph but hopefully I find time to rework all these interpolator classes